--- SSLSocketFactory.java	2002/02/25 20:51:08	1.6
+++ SSLSocketFactory.java	2002/02/28 08:12:36
@@ -40,7 +40,7 @@
 
 //uncomment this line if you want to compile with the separately
 //available jsse1.0.2
-//import com.sun.net.ssl.*;
+import com.sun.net.ssl.*;
 
 public class SSLSocketFactory 
     implements org.jacorb.orb.factory.SocketFactory 
@@ -120,7 +120,7 @@
     {
         //uncomment this line if you want to compile with the separately
         //available jsse1.0.2
-        //Security.addProvider( new com.sun.net.ssl.internal.ssl.Provider() );
+        Security.addProvider( new com.sun.net.ssl.internal.ssl.Provider() );
 
 	try 
 	{
