--- SSLServerSocketFactory.java	2002/02/25 20:51:07	1.6
+++ SSLServerSocketFactory.java	2002/02/28 08:14:51
@@ -11,7 +11,7 @@
 
 //uncomment this line if you want to compile with the separately
 //available jsse1.0.2
-//import com.sun.net.ssl.*;
+import com.sun.net.ssl.*;
 
 import java.net.*;
 import java.io.*;
@@ -33,7 +33,7 @@
     {
         //uncomment this line if you want to compile with the separately
         //available jsse1.0.2
-        //Security.addProvider( new com.sun.net.ssl.internal.ssl.Provider() );
+        Security.addProvider( new com.sun.net.ssl.internal.ssl.Provider() );
 
 	factory = createServerSocketFactory();
 
